src/CSV/FingersCrossed/Loader.php

```diff
@@ -22,28 +22,27 @@
     public function __construct(
         private WriterInterface $writer,
         private LoggerInterface $logger = new NullLogger()
-    ) {
-    }
+    ) {}
 
     public function load(): \Generator
     {
-        $line = yield;
+        $line = yield new EmptyResultBucket();
         try {
             $this->writer->addRow(
                 new Row(array_map(fn ($value) => new Cell($value), array_keys($line)), null)
             );
-        } catch (WriterNotOpenedException|IOException $exception) {
+        } catch (IOException|WriterNotOpenedException $exception) {
             $this->logger->error('Impossible to load data to the given CSV file.', ['line' => $line, 'message' => $exception->getMessage(), 'previous' => $exception->getPrevious()]);
 
             return;
         }
 
-        while (true) {
+        while ($line) {
```

Comment:
Pourquoi est-ce que tu as remplacé `true` par `$line` ?

Reply:
Je l'ai remplacé car phpstan me signale une erreur avec `while condition is always true`

Reply:
Oui, il faut ajouter une exception à PHPStan sur cette ligne, sinon tu modifies le fonctionnement de la pipeline entière.

Exemple: https://github.com/php-etl/pipeline/blob/main/src/Loader/LogLoader.php#L26

Reply:
![image](https://github.com/php-etl/spreadsheet-flow/assets/47692802/023f8569-5e75-4096-8053-0925ac581ce6)